sockeye/translate.py

```diff
@@ -88,7 +89,16 @@ def run_translate(args: argparse.Namespace):
             override_dtype=args.override_dtype,
             output_scores=output_handler.reports_score(),
             sampling=args.sample)
-
+        
+        max_ptr = vocab.get_max_ptr(target_vocab)
+        if max_ptr is not None:
+            check_condition(args.restrict_lexicon is None,
+                            "The pointer networks does not currently work with vocabulary restriction.")
+            
+            for model in models:
+                check_condition(max_ptr >= model.max_supported_seq_len_source,
```

Comment:
So `max_supported_seq_len_source` is an `Optional[int]`. If the model does not have any restriction regarding the source length it is `None`. Therefore the comparison with an int would fail. How about modifying the implementation of `InferenceModel.max_supported_seq_len_source` to return a non-None value in case the model uses pointer networks. This way the maximum length gets set automatically to the maximum and the user is not faced with an error message. What do you think?

Reply:
Good point! I implemented the change to InferenceModel.max_supported_seq_len_source as you suggested. I now notice that in this way the check on max_input_length was already implemented. Thanks!